feat: explain building and releasing test version (#2654)
BTE-Trusted-Entity/sporran-extension: docs/internal.md

1. Go to [actions page for "Packages the internal version"](https://github.com/KILTprotocol/sporran-extension/actions/workflows/publish.yaml).
1. Click on the build you want to test.

> 1. Download the extension file from the _Artifact's_ section at the bottom.

Reviewer: Since the section is called "Artifacts" I think the grammar checker misguided you here. I have my doubts about any apostrophe here, but I am ready to accept the version consistent with the one you have below.
```suggestion
1. Download the extension file from the _Artifacts’_ section at the bottom.
```

Author: I'm just deleting them all. English grammar is not known for being strict.

Author: gone

```

This will generate a file `dist/web-ext-artifacts/sporran-???.zip` ready to be uploaded to stores.
This version will only connect to _KILT Spiritnet_ blockchain via predefined node endpoints.

### For the internal version

Run the following commands:

```
yarn install
yarn build-internal
yarn package
```

This will generate a file `dist/web-ext-artifacts/sporran-???.zip` ready for developers to use.
This version will connect to any node endpoints the user inputs and by default to the _KILT Peregrine_ blockchain.

> ## Uploading to the release's site

Reviewer: ```suggestion
## Uploading to the releases page
```
Unlike in German, in English "site" in relation to the internet pretty much always means everything on a given domain, not a single page

Author: ok. 

changed.

> We recommend external developers to download the test version of the Sporran extension from [the releases' page on GitHub](https://github.com/BTE-Trusted-Entity/sporran-extension/releases).

Reviewer: Weird to see recommendations for external developers in the documentation for internal developers

Author: This is just to explain why do we need to upload it there.

Reviewer: Oh right, that’s fair

The uploads to this page occur manually.
Only Sporran versions using changes merged to `main` should be published as releases.

To get a releasable Sporran Extension file you could either [build and package it locally](#For-the-internal-version) or, alternatively, follow these steps:

1. Merge the wished changes to `main`.
2. Go to the [Sporran GitHub's Actions page for "Packages the internal version"](https://github.com/BTE-Trusted-Entity/sporran-extension/actions/workflows/publish.yaml).
3. Click on the workflow of the wished commit.
4. Download the artifact.

> Before updating it to the release site, please prepend a _"TEST-"_ to the zip file's name.

Reviewer: ```suggestion
Before updating it to the releases page, please prepend a _"TEST-"_ to the zip file's name.
```

Author: implemented

> If you got the extension from the GitHub workflow, it would be nice to replace the commit hash with the sporran version on the zip's name.

Reviewer: ```suggestion
If you got the extension from the GitHub workflow, it would be nice to replace the commit hash with the Sporran version on the zip's name.
```

Author: implemented

To upload:

> 1. Go to [the releases' page](https://github.com/BTE-Trusted-Entity/sporran-extension/releases).

Reviewer: ```suggestion
1. Go to [the releases page](https://github.com/BTE-Trusted-Entity/sporran-extension/releases).
```

Author: The grammar checker told me to add those apostrophes.

Reviewer: There is a nice irony to this :grin:  [The GitHub however doesn’t use them](https://docs.github.com/en/repositories/releasing-projects-on-github/viewing-your-repositorys-releases-and-tags).

Author: Is here ok? 
> _Artifact's_ section

Author: I just saw your new comment below.

Author: https://www.grammarly.com/blog/possessive-apostrophe/

2. Copy title and description of an older release.
3. Click on [Draft a new release](https://github.com/BTE-Trusted-Entity/sporran-extension/releases/new)
4. Paste title and description and attach the Sporran extension zip file.

> 5. Choose as tag the version of Sporran being release.

Reviewer: ```suggestion
5. Choose as tag the version of Sporran being released.
```

Author: thanks. fixed.
